spec/classes/server_spec.rb

```diff
@@ -31,7 +31,7 @@
             :servertype   => server_type,
             :storeconfigs => 'puppetdb',
             :environmentpath => '$confdir/environments',
-            :basemodulepath => '/etc/puppet/modules:/usr/share/puppet/modules',
+            :basemodulepath => ['/etc/puppet/modules', '/usr/share/puppet/modules'],
```

Comment:
It's still unclear to me what I should do: should I just remove the validation? Is it possible to validate if an input is array OR string?

As a side note, I personally believe it is better to accept only arrays. Backwards compatibility is a big topic, and it would make better sense to force it if we were using semver properly here, but that's a big topic.